ProgrammaticUI/Controllers/HomeViewController.swift

```diff
@@ -65,6 +67,39 @@ class HomeViewController: UIViewController {
         navigationController?.pushViewController(insertDataVC, animated: true)
     }
 
+    private func loadData() {
+        guard let userEmail = Auth.auth().currentUser?.email else {
+            fatalError("User not authenticated.")
+        }
+
+        let ref = Database.database().reference().child("users").child(getUsername(from: userEmail))
```

Comment:
I would statically allocate `users`, `km`, `calories` and `duration` as properties in a new struct so they can be commonly accessed here and in `InserDataVC`.